wave_lang/kernel/wave/templates/speculative_decoding.py

```diff
@@ -162,58 +23,84 @@ def get_speculative_sampling_kernel(
     CUR_INDEX = sympy.Symbol("CUR_INDEX")
     J = sympy.Symbol("J")
     BATCH_SIZE = tkl.sym.BATCH_SIZE
+    CUR_PROB_OFFSET = tkl.sym.CUR_PROB_OFFSET
+    DRAFT_TOKEN_ID = tkl.sym.DRAFT_TOKEN_ID
+    LAST_ACCEPTED_RETRIEVE_IDX = tkl.sym.LAST_ACCEPTED_RETRIEVE_IDX
+    LAST_IDX = tkl.sym.LAST_IDX
+    LAST_OFFSET = tkl.sym.LAST_OFFSET
+    NUM_ACCEPTED_TOKENS = tkl.sym.NUM_ACCEPTED_TOKENS
     NUM_DRAFT_TOKENS = tkl.sym.NUM_DRAFT_TOKENS
     NUM_SPECULATIVE_TOKENS = tkl.sym.NUM_SPECULATIVE_TOKENS
     VOCAB_SIZE = tkl.sym.VOCAB_SIZE
     SEQ_LEN = tkl.sym.SEQ_LEN
     BLOCK_BATCH_SIZE = tkl.sym.BLOCK_BATCH_SIZE
     BLOCK_NUM_DRAFT_TOK = tkl.sym.BLOCK_NUM_DRAFT_TOK
+    BLOCK_VOCAB_SIZE = tkl.sym.BLOCK_VOCAB_SIZE
     ADDRESS_SPACE = tkl.sym.ADDRESS_SPACE
     GLOBAL_ADDRESS_SPACE_0 = tkl.sym.GLOBAL_ADDRESS_SPACE
 
     hyperparams = {
-        BLOCK_NUM_DRAFT_TOK: 1,
-        NUM_DRAFT_TOKENS: num_draft_tokens,
-        NUM_SPECULATIVE_TOKENS: num_speculative_tokens,
         ADDRESS_SPACE: SHARED_ADDRESS_SPACE,
         GLOBAL_ADDRESS_SPACE_0: GLOBAL_ADDRESS_SPACE,
         BATCH_SIZE: batch_size,
-        BLOCK_BATCH_SIZE: 1,
-        VOCAB_SIZE: vocab_size,
+        NUM_DRAFT_TOKENS: num_draft_tokens,
+        NUM_SPECULATIVE_TOKENS: num_speculative_tokens,
         SEQ_LEN: seq_len,
+        VOCAB_SIZE: vocab_size,
+        BLOCK_BATCH_SIZE: 1,
+        BLOCK_NUM_DRAFT_TOK: 1,
+        BLOCK_VOCAB_SIZE: vocab_size,
     }
 
     dynamic_symbols = []
 
     constraints: list[tkw.Constraint] = [
         tkw.HardwareConstraint(
             threads_per_wave=64,
+            waves_per_block=(1, 1, 1),
             vector_shapes={
-                NUM_DRAFT_TOKENS: num_draft_tokens,
+                NUM_DRAFT_TOKENS: 0,
```

Comment:
why did these change?

Reply:
Add a comment explaining why this is 0 and a TODO that this has to be modified when we switch to the block scan op.